pkg/cmd/cli/cmd/newbuild.go

```diff
@@ -279,5 +279,10 @@ func transformBuildError(err error, fullName, commandPath string, groups errorGr
 		)
 		return
 	}
+	switch err {
+	case newcmd.ErrNoInputs:
+		groups.Add("", "", usageError(commandPath, newBuildNoInput, fullName))
+		return
+	}
```

Comment:
the first switch is checking by the type (types that implements error interface), the newcmd.ErrNoInputs is a built-in error so needs to be checked by the content.

Actually a if resolve but I made this way to be equal the [newapp transformError](https://github.com/openshift/origin/blob/master/pkg/cmd/cli/cmd/newapp.go#L592).

If you prefer I change to a if statement.